content/documentation/4-batch-developer-guides/2-batch/2-spring-batch.md

```diff
@@ -8,34 +8,310 @@ description: 'Create a Spring Batch Job'
+
+    ```java
+    {/* highlight-range{2-3} */}
+    @Configuration
+    @EnableTask
+    @EnableBatchProcessing
+    public class BillingConfiguration {
+      private static final Log logger = LogFactory.getLog(BillingConfiguration.class);
+
+      @Autowired
+      public JobBuilderFactory jobBuilderFactory;
+
+      @Autowired
+      public StepBuilderFactory stepBuilderFactory;
+
+      @Value("${usage.file.name:classpath:usageinfo.json}")
+      private Resource usageResource;
+
+      @Bean
+      public Job job1(ItemReader<Usage> reader,
+        ItemProcessor<Usage,Bill> itemProcessor, ItemWriter<Bill> writer) {
+          Step step = stepBuilderFactory.get("BillProcessing")
+                  .<Usage, Bill>chunk(1)
+                  .reader(reader)
+                  .processor(itemProcessor)
+                  .writer(writer)
+                  .build();
+
+          return jobBuilderFactory.get("BillJob")
+                  .incrementer(new RunIdIncrementer())
+                  .start(step)
+                  .build();
+      }
+
+      @Bean
+      public JsonItemReader<Usage> jsonItemReader() {
+
+          ObjectMapper objectMapper = new ObjectMapper();
+          JacksonJsonObjectReader<Usage> jsonObjectReader =
+                  new JacksonJsonObjectReader<>(Usage.class);
+          jsonObjectReader.setMapper(objectMapper);
+
+          return new JsonItemReaderBuilder<Usage>()
+                  .jsonObjectReader(jsonObjectReader)
+                  .resource(usageResource)
+                  .name("UsageJsonItemReader")
+                  .build();
+      }
+
+      @Bean
+      public ItemWriter<Bill> jdbcBillWriter(DataSource dataSource) {
+          JdbcBatchItemWriter<Bill> writer = new JdbcBatchItemWriterBuilder<Bill>()
+                          .beanMapped()
+                  .dataSource(dataSource)
+                  .sql("INSERT INTO BILL_STATEMENTS (id, first_name, " +
+                     "last_name, minutes, data_usage,bill_amount) VALUES " +
+                     "(:id, :firstName, :lastName, :minutes, :dataUsage, " +
+                     ":billAmount)")
+                  .build();
+          return writer;
+      }
+
+      @Bean
+      ItemProcessor<Usage, Bill> billProcessor() {
+          return new BillProcessor();
+      }
+    }
+    ```
+
+    The `@EnableBatchProcessing` annotation enables Spring Batch features and provide a base configuration for setting up batch jobs.
+    The `@EnableTask` annotation sets up a TaskRepository which stores information about the task execution such as the start and end time of the task and the exit code.
 
 ### Testing
 
-Show a simple test.
+Now let’s create our test. Update the [BillrunApplicationTests.java](https://github.com/spring-cloud/spring-cloud-dataflow-samples/blob/master/dataflow-website/batch-developer-guides/batch/batchsamples/billrun/src/test/java/io/spring/billrun/BillrunApplicationTests.java) such that looks like the contents below.
```

Comment:
Same comment as before about not providing enough detail.  Just telling the user to copy and paste code without telling them what that code does misses the point.